packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -0,0 +1,77 @@
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as cdk from '@aws-cdk/core';
+import { ProductStackSynthesizer } from './private/product-stack-synthesizer';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * A Service Catalog product stack, which is similar in form to a Cloudformation nested stack.
+ * You can add the resources to this stack that you want to define for your service catalog product.
+ *
+ * This stack will not be treated as an independent deployment
+ * artifact (won't be listed in "cdk list" or deployable through "cdk deploy"),
+ * but rather only synthesized as a template and uploaded as an asset to S3.
+ *
+ */
+export class ProductStack extends cdk.Stack {
+  public readonly templateFile: string;
```

Comment:
Is there any reason this is `public`?

Reply:
Yes, because it's public member in `stack.ts`

https://github.com/aws/aws-cdk/blob/1ab9b265e9899ffcd093b3600d658c8a6519cc69/packages/%40aws-cdk/core/lib/stack.ts#L285

Reply:
OK. That's all good then, leave it as-is.